### PR TITLE
Large Model Experimentation

### PR DESCRIPTION
**Pull request by s-maddrellmander**
Updates from development branch, some details include adding epochs and samples seen to metrics to track experiments on wandb x axis, and formatting the filename of the checkpoints which are saved to be uploaded to wandb to use the date time.

### CODE REVIEW

File: graphium/trainer/predictor.py [hunk 1]
```diff
@@ -378,6 +384,9 @@ def _general_step(self, batch: Dict[str, Tensor], step_name: str, to_cpu: bool)
         # print("loss ", self.global_step, self.current_epoch, loss)
         step_dict["task_losses"] = task_losses
         step_dict["gradient_norm"] = self.get_gradient_norm()
+        # samuelm
+        # self.samples_seen += 1
+        # step_dict["samples_seen"] = self.samples_seen
```
Comment (s-maddrellmander) [inline graphium/trainer/predictor.py:389]:
```suggestion
```

File: graphium/trainer/predictor.py [hunk 2]
```diff
@@ -447,6 +456,9 @@ def flag_step(self, batch: Dict[str, Tensor], step_name: str, to_cpu: bool) -> D
         step_dict[f"loss/{step_name}"] = loss.detach().cpu()
         step_dict["loss"] = loss
         step_dict["task_losses"] = task_losses
+        # samuelm
+        # self.samples_seen += 1
+        # step_dict["samples_seen"] = self.samples_seen
```
Comment (s-maddrellmander) [inline graphium/trainer/predictor.py:461]:
```suggestion
```
